tests/functional/s3/test_cp_command.py

```diff
@@ -615,7 +616,29 @@ def test_upload_unicode_path(self):
         progress_message = 'Completed 10 Bytes'
         self.assertIn(progress_message, stdout)
 
-    def test_cp_with_error_and_warning(self):
+
+    @skip_if_windows("Unreadable file is hard to make on windows")
+    def test_cp_with_error_and_warning_unix(self):
+        command = "s3 cp --recursive %s s3://bucket/"
+        self.parsed_responses = [{
+            'Error': {
+                'Code': 'NoSuchBucket',
+                'Message': 'The specified bucket does not exist',
+                'BucketName': 'bucket'
+            }
+        }]
+        self.http_response.status_code = 404
+
+        unreadable_path = self.files.create_file('foo.txt', 'foo')
+        self.files.create_file('bar.txt', 'bar')
+        os.chmod(unreadable_path, 0o222)
+        _, stderr, rc = self.run_cmd(command % self.files.rootdir, expected_rc=1)
+        self.assertIn('upload failed', stderr)
+        self.assertIn('warning: Skipping file', stderr)
+        self.assertIn('File/Directory is not readable.', stderr)
+
+    @skip_if_macos("On MacOS we cannot create an invalid timestamp.")
+    def test_cp_with_error_and_warning_windows(self):
```

Comment:
Maybe we should name it `test_cp_with_error_and_invalid_timestamp_warning`? Mainly this test will run for Linux as well. We can then use a similar naming approach in the previous test for unreadable files.